### DISCUSSION TITLE
Custom CNN output is not correct

### DISCUSSION THREAD

**Discussion by noahzn** [bitcraze/discussions#579] [Q&A]
Hi, I already deployed my CNN on the AI deck and it can run and print something through the cpx. However, the output shown in the cfclient is not correct. 

My output of the CNN is an array with 16 numbers. I verified the output in a dumped .npy file, and it's correct. As shown in the picture below, the array with 2 elements is the output of the official AI deck classification demo, and the 16-elements array is my output. 
![image](https://user-images.githubusercontent.com/8244850/220863203-bec88e2e-074e-41b3-9678-0c202e12b160.png)

In the main .c file, I use a for loop to print `cpxPrintToConsole(LOG_TO_CRTP, "%d, %hd\n", i, Output_1[i])`, (0<=i<=15). The type of Output_1 is `signed short`, but what I got in the console is 
```
0, 12849
1, 13365
2, 12595
....
10, -14409
11, 7174
12, -31341
....
```

My questions are: 1), why are there some negative numbers? The numbers in the array are all positive numbers. 2),My network receives 1-channel images, do I need to change anything in the .c file? The default classification.c has this line '#define IO RGB888_IO'. 3). I found in the auto-generated AT model the `Output_1` of the classification CNN is `signed short`, but in my auto-generated AT model, it is defined as `'signed char'`. 4), Is there an easier way to debug? Now everytime I rebuild the program and use cfloader to deploy it on AI deck. Thanks!

**Comment by knmcguire** [comment]
That seems quite off indeed... Perhaps you could cut it up in some smaller pieces first:

1- Test out the full NN on your computer (you have done that)
2- Test out the TF light model
3- Test out the quantized model (by NNtools?)
4- Test out on the AIdeck.

There is also a gap8 simulated chip environment, of which they have some examples of that on their repo. You'll need to remove any crazyflie/cpx related stuff though but you should be able to feed it images from there and try it out on your computer. I think more information should be in the GAP8 documentation of greenwaves.
